lxd/storage/drivers/driver_dir.go

```diff
@@ -208,8 +209,28 @@ func (d *dir) CreateVolume(vol Volume, filler func(path string) error, op *opera
 		return err
 	}
 
+	// Create sparse loopback file if volume is block.
+	rootBlockPath := ""
+	if vol.contentType == ContentTypeBlock {
+		rootBlockPath, err = d.GetVolumeDiskPath(vol.volType, vol.name)
+		if err != nil {
+			return err
+		}
+
+		// tomp TODO what size should this be?
+		sizeBytes, err := units.ParseByteSizeString("10GB")
```

Comment:
@stgraber currently I am creating a 10GB sparse raw image for the VM image to be unpacked into. Should this take a size from the quota size and if not specified default to a fixed value? 

Also, for dir backend, should I still initialise the project quota on the dir that contains the sparse file as well?

Reply:
As per above, when storing as a file, we should store as qcow2, not raw, so effectively copying the source image over, then doing a resize operation on the qcow2 to match the quota.

I don't think we want to apply the project quotas in this case as the qcow2 size is the ultimate limit there. That reminds me, we shouldn't have the snapshots be size restricted in general in the storage API, so we should update `dir` to clear any quota on the snapshot. Otherwise creating snapshots would grow your total size very quickly.

As for the size of the image to unpack into, for non-block, we just copy the qcow2 and attempt to grow it up to the limit we have, if no limit, we don't touch it. For blocks, we'll use the same logic we have right now, effectively looking at `volume.size` from the pool and falling back to 10GB. Unpack the image in there, then clone the volume for the vm and grow that to the requested size for the vm.

Reply:
Looking at the dir snapshot quota case, I think we're good, assuming that rsync doesn't know how to copy those too (and I doubt that it does).

Reply:
@stgraber has fixed this in PR https://github.com/lxc/lxd/pull/6417